impl/src/main/java/com/okta/sdk/impl/oauth2/OAuth2ClientCredentials.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.oauth2;
+
+import com.okta.commons.lang.Assert;
+import com.okta.sdk.authc.credentials.ClientCredentials;
+
+public class OAuth2ClientCredentials implements ClientCredentials<OAuth2AccessToken> {
+
+    private OAuth2AccessToken oAuth2AccessToken;
+    private AccessTokenRetrieverService accessTokenRetrieverService;
+
+    public OAuth2ClientCredentials(OAuth2AccessToken oAuth2AccessToken,
+                                   AccessTokenRetrieverService accessTokenRetrieverService) {
```

Comment:
This seems odd, you have both a token and a retriever service? My guess is you should just pass in the service and get the token from there?